pycroft/lib/membership.py

```diff
@@ -185,3 +189,25 @@ def user_memberships_query(user_id: int, active_groups_only: bool = False):
             .group_by(Membership.id)
     )
     return memberships
+
+
+def change_membership_active_during(
```

Comment:
Add `with_transaction` above.

Reply:
-1, we will deprecate this decorator at some point. Altering the session lifecycle should not be done in the lib. See [this discussion](https://github.com/agdsn/pycroft/issues/136#issuecomment-899034398). It makes things harder to control / reason about.
Perhaps I should add an ADR explaining the details.

Reply:
Actually this is sufficiently commented in [ADR004](https://agdsn.github.io/pycroft/arch/adr-004.html#cost-vs-benefit). The real thing missing here is to put `session` as the first parameter, but that wouldn't make any sense as of yet because `log_user_event` does not use the session explicitly.
So I guess one should fix it at another time, switching the paradigm for more than just one method.